File: original artifact/Ian_Foss_cs-360_project_three_inventory_app/cs360_project_three/app/src/main/java/com/example/cs_360_project_three/UserDao.java
package com.example.cs_360_project_three;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.cs_360_project_three.UserEntity;
@Dao
public interface UserDao {

    @Insert
    long insertUser(UserEntity user);


    // Get one user that matches username
    @Query("SELECT * FROM user_table WHERE username = :username LIMIT 1")
    UserEntity getUserByUsername(String username);


    // Query used to validate login credentials
    @Query("SELECT * FROM user_table WHERE username = :username AND password = :password LIMIT 1")
    UserEntity getUserByCredentials(String username, String password);


    // Query to see if username already exists
    @Query("SELECT COUNT(*) FROM user_table WHERE username = :username")
    int usernameExists(String username);


    @Update
    void updateUser(UserEntity user);

    @Delete
    void deleteUser(UserEntity user);



}

File: enhancements/enhancement-one/Enhancement One/app/src/main/java/com/example/CS499_Capstone/UserDao.kt
package com.example.CS499_Capstone

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface UserDao {
    @Insert
    fun insertUser(user: UserEntity): Long


    // Get one user that matches username
    @Query("SELECT * FROM user_table WHERE username = :username LIMIT 1")
    fun getUserByUsername(username: String): UserEntity


    // Query used to validate login credentials
    @Query("SELECT * FROM user_table WHERE username = :username AND password = :password LIMIT 1")
    fun getUserByCredentials(username: String, password: String): UserEntity?


    // Query to see if username already exists
    @Query("SELECT COUNT(*) FROM user_table WHERE username = :username")
    fun usernameExists(username: String): Int


    @Update
    fun updateUser(user: UserEntity)

    @Delete
    fun deleteUser(user: UserEntity)
}


